### DISCUSSION TITLE
Optional Feature Request (but I am not sure if it would be sufficiently useful) - mention the executables that will be installed per gem

### DISCUSSION THREAD

**Discussion by ghost** [rubygems/rubygems.org#3352] [Ideas]
Hello ruby hackers, in particular rubygems.org maintainers and devs,

Lately I changed some of my gems to include a file under bin/ subdirectory, an executable.

For example, if my project would be fairly large, called "foobar", then I thought that
it would be nice if users could do "foobar --help" consistently to get some help or
documentation on how to use that project. Note that this is mostly for my larger
gems, the smaller gems are usually so small that adding a bin/foobar would
almost make them twice as big. :D

Anyway, so as I was doing that change, I wondered:

**- Should there be a simple way to somehow find out (and thus display) which
bin/ executable files are installed, somewhere, for any particular gem?**

I thought that this information may be useful to have BUT I do not really know
where this could be displayed, so that it does not potentially distract from other
content on a rubygems.org page for a particular gem.

So I guess before this could even get to the process of being reviewed and 
added one day, it should first be decided if this is even worth to have; and
if so, afterwards, how or where this gets shown or displayed.

The information should already be stored within a gem / .gemspec file
at the least, if it was written properly, in particular via the:

**.executables=**

entry in a gemspec file.

So we can assume this information to be there, in a proper gem, and now
it should be decided whether this would be useful (it sure enough isn't 
that important to many people, so this should be optional in all way in
its display; I guess most people who download gems will not care, but
perhaps a few may care, so optionally this may be useful to these people).

Let us look at the popular gem called **rake**:

https://rubygems.org/gems/rake

In its bin/ subdirectory it has these two executables:

**console
setup**

So, console and setup.

How could we show this information on the above webpage?

Well... I guess the main page with "Versions" and "Development Dependencies"
is too important so we can't add or should not add more information to that.

That leaves the right side pane, which already contains quite some information,
such as "Badge", "Subscribe" or "RSS".

How about adding an entry there such as "Executables"? This one would be
optional, that is, only displayed IF that particular gem actually HAS such
an Executable. An alternative may be to add an extra page to gems showing
a lot more information in general, and then displaying it there - but I don't
want to make this suggestion too complicated and rather want to keep it
simple. The query API could include a handler such as "/executables" or
"/executables?" as extra command to a particular gem-version or something
like that.

Anyway, I hope my suggestion is understandable - if you think that this 
suggestion is not useful to have, feel free to close it at any moment in 
time to keep the issue-tracker for more important issues. Thanks!

**Comment by simi** [comment]
This seems similar to #3349, feel free to follow recommended steps at https://github.com/rubygems/rubygems.org/discussions/3349#discussioncomment-4712203 to move this forward. Moving to discussions for now.
